hummingbot/connector/exchange/kucoin/kucoin_market.pyx

```diff
@@ -411,44 +411,27 @@ cdef class KucoinMarket(ExchangeBase):
                 raise IOError(f"Error parsing data from {url}.")
             return parsed_response
 
-    async def _update_account_id(self) -> str:
-        accounts = await self._api_request("get", path_url="/api/v1/accounts", is_auth_required=True)
-        try:
-            for account in accounts["data"]:
-                if account["type"] == "trade":
-                    self._account_id = str(account["id"])
-        except Exception as e:
-            raise ValueError(f"Unable to retrieve account id: {e}")
-
     async def _update_balances(self):
         cdef:
-            str path_url = "/api/v1/accounts"
-            dict data = await self._api_request("get", path_url=path_url, is_auth_required=True)
-            list balances = data.get("list", [])
-            dict new_available_balances = {}
-            dict new_balances = {}
+            str path_url = "/api/v1/accounts?type=trade"
             str asset_name
-            object balance
+            set local_asset_names = set(self._account_balances.keys())
+            set remote_asset_names = set()
 
+        data = await self._api_request("get", path_url=path_url, is_auth_required=True)
         if data:
             for balance_entry in data["data"]:
                 asset_name = balance_entry["currency"]
-                balance = Decimal(balance_entry["balance"])
-                if balance == s_decimal_0:
-                    continue
-                if asset_name not in new_available_balances:
-                    new_available_balances[asset_name] = s_decimal_0
-                if asset_name not in new_balances:
-                    new_balances[asset_name] = s_decimal_0
-
-                new_balances[asset_name] += balance
-                if balance_entry["type"] == "trade":
-                    new_available_balances[asset_name] = Decimal(balance_entry["available"])
+                new_balances = Decimal(balance_entry["balance"])
+                new_available_balances = Decimal(balance_entry["available"])
+                self._account_available_balances[asset_name] = new_available_balances
```

Comment:
This would be simpler with
`self._account_available_balances[asset_name] = Decimal(balance_entry["available"])`
as `new_balances` is a new variable without cdef declaration (so it will be slow), and the name is confusing as it is not plural.